specification/marketplacecatalog/resource-manager/Solutions/preview/2023-03-01-preview/solutions.json

```diff
@@ -0,0 +1,501 @@
+    },
+    "ApiVersionParameter": {
+      "name": "api-version",
+      "in": "query",
+      "required": true,
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "description": "The API version to use for the request."
+    },
+    "ProductIdPathParameter": {
+      "in": "path",
+      "name": "productId",
+      "type": "string",
+      "x-ms-parameter-location": "method",
+      "required": true,
+      "description": "The product id to update or delete"
+    }
+  },
+  "definitions": {
+    "CreateOrUpdateSolutionRequestPayload": {
+      "type": "object",
+      "properties": {
+        "solutionName": {
+          "type": "string"
+        },
+        "productIds": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "additionalProperties": false
+    },
+    "AddProductToUserSolutionRequest": {
+      "type": "object",
+      "properties": {
+        "productId": {
+          "type": "string"
+        }
+      },
+      "additionalProperties": false
+    },
+    "Product": {
+      "type": "object",
+      "properties": {
+        "uniqueProductId": {
+          "type": "string"
+        },
+        "displayName": {
+          "type": "string"
+        },
+        "description": {
+          "type": "string"
+        },
+        "summary": {
+          "type": "string"
+        },
+        "smallIconUri": {
+          "type": "string"
+        },
+        "publisherDisplayName": {
+          "type": "string"
+        },
+        "productType": {
+          "type": "string"
+        },
+        "pricingTypes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "storeFronts": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "ratingAverage": {
+          "type": "number",
+          "format": "double"
+        }
+      },
+      "additionalProperties": false
+    },
+    "RenameUserSolutionRequest": {
+      "description": "The service properties for a given service.",
+      "type": "object",
+      "properties": {
+        "newName": {
+          "type": "string"
+        }
+      }
+    },
+    "Solution": {
+      "description": "The Solution data structure",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/SolutionProperties",
+          "description": "The Solution data structure",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "SolutionProperties": {
+      "description": "Describes the json payload for solution operations",
+      "type": "object",
+      "properties": {
+        "solutionId": {
+          "type": "string"
+        },
+        "solutionName": {
+          "type": "string"
+        },
+        "products": {
```

Comment:
>products [](http://example.com/codeflow?start=9&length=8)

products should either be operated on as an array inside of a solution OR have their own child resource type where they are operated on.  Right now this spec has both which is not supported

Reply:
We want the user to perform GET/DELETE operations on products inside some solution using only the productId, but Solution response should return an extended Product schema

Reply:
done